### PR TITLE
fix: type validation

### PR DESCRIPTION
**Pull request by chloeminkyung**
fix: type validation

### CODE REVIEW

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 1]
```diff
@@ -311,10 +424,104 @@ macro_rules! define_comparison {
...
+                Field::Timestamp(left_v) => match right_p {
+                    Field::Timestamp(right_v) => Ok(Field::Boolean($function(left_v, right_v))),
+                    Field::String(right_v) | Field::Text(right_v) => {
+                        let right_val = DateTime::<FixedOffset>::from_str(right_v.as_str())
+                            .map_err(|_| {
+                                PipelineError::UnableToCast(
+                                    format!("{}", right_v),
+                                    "Timestamp".to_string(),
+                                )
+                            })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:521]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 2]
```diff
@@ -333,9 +539,14 @@ macro_rules! define_comparison {
...
                 },
-                Field::Text(ref left_v) => match right_p {
-                    Field::String(ref right_v) => Ok(Field::Boolean($function(left_v, right_v))),
-                    Field::Text(ref right_v) => Ok(Field::Boolean($function(left_v, right_v))),
+                Field::Date(left_v) => match right_p {
+                    Field::Date(right_v) => Ok(Field::Boolean($function(left_v, right_v))),
+                    Field::String(right_v) | Field::Text(right_v) => {
+                        let right_val = NaiveDate::from_str(right_v.as_str()).map_err(|_| {
+                            PipelineError::UnableToCast(format!("{}", right_v), "Date".to_string())
+                        })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:547]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 3]
```diff
@@ -690,10 +982,93 @@ pub fn evaluate_lt(
...
+                let left_val = Decimal::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Decimal".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::Timestamp(right_v) => {
+                let left_val: DateTime<FixedOffset> = DateTime::<FixedOffset>::from_str(left_v)
+                    .map_err(|_| {
+                        PipelineError::UnableToCast(format!("{}", right_v), "Timestamp".to_string())
+                    })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1035]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 4]
```diff
@@ -690,10 +982,93 @@ pub fn evaluate_lt(
...
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::Date(right_v) => {
+                let left_val = NaiveDate::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Date".to_string())
+                })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1041]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 5]
```diff
@@ -690,10 +982,93 @@ pub fn evaluate_lt(
...
+                "<".to_string(),
+            )),
+        },
+        Field::Timestamp(left_v) => match right_p {
+            Field::Timestamp(right_v) => Ok(Field::Boolean(left_v < right_v)),
+            Field::String(right_v) | Field::Text(right_v) => {
+                let right_val: DateTime<FixedOffset> =
+                    DateTime::<FixedOffset>::from_str(right_v.as_str()).map_err(|_| {
+                        PipelineError::UnableToCast(right_v.to_string(), "Timestamp".to_string())
+                    })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1068]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 6]
```diff
@@ -712,9 +1086,14 @@ pub fn evaluate_lt(
...
         },
-        Field::Text(ref left_v) => match right_p {
-            Field::String(ref right_v) => Ok(Field::Boolean(left_v < right_v)),
-            Field::Text(ref right_v) => Ok(Field::Boolean(left_v < right_v)),
+        Field::Date(left_v) => match right_p {
+            Field::Date(right_v) => Ok(Field::Boolean(left_v < right_v)),
+            Field::String(right_v) | Field::Text(right_v) => {
+                let right_val = NaiveDate::from_str(right_v.as_str()).map_err(|_| {
+                    PipelineError::UnableToCast(right_v.to_string(), "Date".to_string())
+                })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1094]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 7]
```diff
@@ -1067,10 +1518,93 @@ pub fn evaluate_gt(
...
+                let left_val: DateTime<FixedOffset> = DateTime::<FixedOffset>::from_str(left_v)
+                    .map_err(|_| {
+                        PipelineError::UnableToCast(format!("{}", right_v), "Timestamp".to_string())
+                    })?;
+                Ok(Field::Boolean(left_val > right_v))
+            }
+            Field::Date(right_v) => {
+                let left_val = NaiveDate::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Date".to_string())
+                })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1577]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 8]
```diff
@@ -1067,10 +1518,93 @@ pub fn evaluate_gt(
...
+                ">".to_string(),
+            )),
+        },
+        Field::Timestamp(left_v) => match right_p {
+            Field::Timestamp(right_v) => Ok(Field::Boolean(left_v > right_v)),
+            Field::String(right_v) | Field::Text(right_v) => {
+                let right_val: DateTime<FixedOffset> =
+                    DateTime::<FixedOffset>::from_str(right_v.as_str()).map_err(|_| {
+                        PipelineError::UnableToCast(right_v.to_string(), "Timestamp".to_string())
+                    })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1604]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

File: dozer-sql/src/pipeline/expression/comparison.rs [hunk 9]
```diff
@@ -1089,9 +1622,14 @@ pub fn evaluate_gt(
...
         },
-        Field::Text(ref left_v) => match right_p {
-            Field::String(ref right_v) => Ok(Field::Boolean(left_v > right_v)),
-            Field::Text(ref right_v) => Ok(Field::Boolean(left_v > right_v)),
+        Field::Date(left_v) => match right_p {
+            Field::Date(right_v) => Ok(Field::Boolean(left_v > right_v)),
+            Field::String(right_v) | Field::Text(right_v) => {
+                let right_val = NaiveDate::from_str(right_v.as_str()).map_err(|_| {
+                    PipelineError::UnableToCast(right_v.to_string(), "Date".to_string())
+                })?;
```
Comment (mediuminvader) [inline dozer-sql/src/pipeline/expression/comparison.rs:1630]: I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

**Review by snork-alt — APPROVED** [review]

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 9 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 8 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 7 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 6 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 5 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 4 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 3 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 2 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

File: dozer-sql/src/pipeline/expression/comparison.rs (hunk 1 above)
Comment (chloeminkyung) [reply dozer-sql/src/pipeline/expression/comparison.rs]: fixed

**Pull request merged**